x-pack/plugins/ml/public/alerting/job_selector.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiComboBox, EuiComboBoxOptionOption, EuiComboBoxProps, EuiFormRow } from '@elastic/eui';
+import { JobId } from '../../common/types/anomaly_detection_jobs';
+import { MlApiServices } from '../application/services/ml_api_service';
+
+interface JobSelection {
+  jobIds?: JobId[];
+  groupIds?: string[];
+}
+
+export interface JobSelectorControlProps {
+  jobSelection?: JobSelection;
+  onSelectionChange: (jobSelection: JobSelection) => void;
+  adJobsApiService: MlApiServices['jobs'];
+  /**
+   * Validation is handled by alerting framework
+   */
+  errors: string[];
+}
+
+export const JobSelectorControl: FC<JobSelectorControlProps> = ({
+  jobSelection,
+  onSelectionChange,
+  adJobsApiService,
+  errors,
+}) => {
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const jobIds = useMemo(() => new Set(), []);
+  const groupIds = useMemo(() => new Set(), []);
+
+  const fetchOptions = useCallback(async () => {
+    try {
+      const {
+        jobIds: jobIdOptions,
+        groupIds: groupIdOptions,
+      } = await adJobsApiService.getAllJobAndGroupIds();
+
+      jobIdOptions.forEach((v) => {
+        jobIds.add(v);
+      });
+      groupIdOptions.forEach((v) => {
+        groupIds.add(v);
+      });
+
+      setOptions([
+        {
+          label: i18n.translate('xpack.ml.jobSelector.jobOptionsLabel', {
+            defaultMessage: 'Jobs',
+          }),
+          options: jobIdOptions.map((v) => ({ label: v })),
+        },
+        {
+          label: i18n.translate('xpack.ml.jobSelector.groupOptionsLabel', {
+            defaultMessage: 'Groups',
+          }),
+          options: groupIdOptions.map((v) => ({ label: v })),
+        },
+      ]);
+    } catch (e) {
+      // TODO add error handling
```

Comment:
Is this planned for a follow-up?

Reply:
Yes, I need to think about this scenario (which is pretty uncommon I believe)